src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/PyrisConnectorService.java

```diff
@@ -95,15 +114,100 @@ public void executePipeline(String feature, String variant, Object executionDTO)
      * @param variant      The variant of the feature to execute
      * @param executionDTO The DTO sent as a body for the execution
      */
-    public void executeLectureWebhook(String variant, PyrisWebhookLectureIngestionExecutionDTO executionDTO) {
+    public void executeLectureAddtionWebhook(String variant, PyrisWebhookLectureIngestionExecutionDTO executionDTO) {
         var endpoint = "/api/v1/webhooks/lectures/" + variant;
         try {
             restTemplate.postForEntity(pyrisUrl + endpoint, objectMapper.valueToTree(executionDTO), Void.class);
         }
+        catch (HttpStatusCodeException e) {
+            log.error("Failed to send lecture unit {} to Pyris: {}", executionDTO.pyrisLectureUnit().lectureUnitId(), e.getMessage());
+            throw toIrisException(e);
+        }
+        catch (RestClientException | IllegalArgumentException e) {
+            log.error("Failed to send lecture unit {} to Pyris: {}", executionDTO.pyrisLectureUnit().lectureUnitId(), e.getMessage());
+            throw new PyrisConnectorException("Could not fetch response from Pyris");
+        }
+    }
+
+    /**
+     * uses getLectureUnitIngestionState for all lecture units and then determines the IngestionState of the lecture
+     *
+     * @param courseId  id of the course
+     * @param lectureId id of the lecture
+     * @return The ingestion state of the lecture
+     *
+     */
+    public IngestionState getLectureIngestionState(long courseId, long lectureId) {
+        try {
+            List<LectureUnit> lectureunits = lectureRepository.findByIdWithLectureUnitsAndAttachments(lectureId).get().getLectureUnits();
```

Comment:
```suggestion
            List<LectureUnit> lectureunits = lectureRepository.findByIdWithLectureUnits(lectureId).get().getLectureUnits();
```
I don't think you need the attachments here, do you? You could even add a method in lectureUnitRepository to only retrieve the lecture units and not the lecture